storage/queuesasuri.go

```diff
@@ -0,0 +1,102 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// GetSASURIWithSignedIP creates an URL to the specified queue which contains the Shared
+// Access Signature with specified permissions and expiration time.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/ee395415.aspx
+func (q *Queue) GetSASURIWithSignedIP(expiry time.Time, permissions string, signedIPRange string) (string, error) {
+	canonicalizedResource, err := q.qsc.client.buildCanonicalizedResource(q.buildPath(), q.qsc.auth)
+	if err != nil {
+		return "", err
+	}
+
+	// "The canonicalizedresouce portion of the string is a canonical path to the signed resource.
+	// It must include the service name (blob, table, queue or file) for version 2015-02-21 or
+	// later, the storage account name, and the resource name, and must be URL-decoded.
+	// -- https://msdn.microsoft.com/en-us/library/azure/dn140255.aspx
+	// We need to replace + with %2b first to avoid being treated as a space (which is correct for query strings, but not the path component).
+	canonicalizedResource = strings.Replace(canonicalizedResource, "+", "%2b", -1)
+	canonicalizedResource, err = url.QueryUnescape(canonicalizedResource)
+	if err != nil {
+		return "", err
+	}
+
+	// assumption that start time is now.
+	//signedStart := time.Now().UTC().Format(time.RFC3339)
+	signedExpiry := expiry.UTC().Format(time.RFC3339)
+
+	// Cannot get this working yet. Any values entered generates bad URL.
+	protocols := ""
```

Comment:
Updated PR to follow similar lines to @mcardosos BlobSAS changes....  although I still cannot get protocol to work properly.